Add new HASH/BP oriented glossary definitions (#625)
hashintel/hash: resources/glossary/entities.mdx

---
objectId: 5f7f8d88-0205-2222-f46f-0ba3791t4616
title: Entities
description: "Entities are individual ‘things’ with a distinct, independent existence."
slug: entities
tags: ["Graphs", "Simulation Modeling"]
---

## What are entities?

Entities are individual ‘things’ with a distinct, independent existence.

Every entity in HASH has:
- a HASH ID, which makes it uniquely identifiable and content-addressable;
- one or more [entity types](https://hash.ai/glossary/entity-type) which describe the entity’s expected properties;
- [properties](https://hash.ai/glossary/properties), infered from its entity type(s), which may in turn contain [values](https://hash.ai/glossary/values).

The ‘entity type’ tells you what kind of thing an entity is expected to be (e.g. a `Book`, which is expected to have properties such as `author`, `title` and `publisher`).

The values of an entity’s properties tell you about that instance of the entity itself, for example what its `title` actually is (e.g. “Brave New World”).

> Sometimes the values of a property may contain [links](https://hash.ai/glossary/links). In such cases, we would describe an entity as a ‘linked entity’.

Author: To confirm: property's values will never now contain links?

> Sometimes the values of a property may contain [links](https://hash.ai/glossary/links). In such cases, we would describe an entity as a ‘linked entity’.

Reviewer: Correct, in current designed properties themselves will not be, or have, links.

Links are a first-class concept of things between entities. They have a URI and a name, but that is not the same as a JSON path (i.e. key) within the object

> Sometimes the values of a property may contain [links](https://hash.ai/glossary/links). In such cases, we would describe an entity as a ‘linked entity’.

Reviewer: Correct, in current designed properties themselves will not be, or have, links.

Links are a first-class concept of things between entities. They have a URI and a name, but that is not the same as a JSON path (i.e. key) within the object

> Sometimes the values of a property may contain [links](https://hash.ai/glossary/links). In such cases, we would describe an entity as a ‘linked entity’.

Author: ```suggestion
Separate from its properties, an entity may also have [links](https://hash.ai/glossary/links). In such cases, we would describe an entity as a ‘linked entity’.
```

@Alfred-Mountfield Is this more accurate in light of the new RFC(s)?

> Sometimes the values of a property may contain [links](https://hash.ai/glossary/links). In such cases, we would describe an entity as a ‘linked entity’.

Author: Notwithstanding your prior noted objection to describing such entities as linked entities 🥲
